JOSESwift/Sources/CryptoImplementation/Thumbprint.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  SHA256.swift
+//  JOSESwift
+//
+//  Created by Thomas Torp on 08.06.20.
+//
+//  ---------------------------------------------------------------------------
+//  Copyright 2019 Airside Mobile Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  ---------------------------------------------------------------------------
+//
+
+import Foundation
+import CommonCrypto
+
+enum ThumbprintError: Error {
+    case inputMustBeGreaterThanZero
+}
+
+fileprivate extension ThumbprintAlgorithm {
+    var outputLenght: Int {
+        switch self {
+        case .SHA256:
+            return Int(CC_SHA256_DIGEST_LENGTH)
+        }
+    }
+
+    func calculate(input: UnsafeRawBufferPointer, output: UnsafeMutablePointer<UInt8>) {
+        switch self {
+        case .SHA256:
+            CC_SHA256(input.baseAddress, CC_LONG(input.count), output)
```

Comment:
Would it make sense to check the return type here to make sure the hash was calculated properly?

Reply:
I have never actually validated the return value from CC_SHA256.
Based on the man page it just return the pointer passed in as the md parameter. 
```
extern unsigned char *
     CC_SHA256(const void *data, CC_LONG len, unsigned char *md);
```